lib/api/bucketDeleteReplication.js

```diff
@@ -0,0 +1,56 @@
+const metadata = require('../metadata/wrapper');
+const { metadataValidateBucket } = require('../metadata/metadataUtils');
+const { pushMetric } = require('../utapi/utilities');
+const collectCorsHeaders = require('../utilities/collectCorsHeaders');
+
+/**
+ * bucketDeleteReplication - Delete the bucket replication configuration
+ * @param {AuthInfo} authInfo - Instance of AuthInfo class with requester's info
+ * @param {object} request - http request object
+ * @param {object} log - Werelogs logger
+ * @param {function} callback - callback to server
+ * @return {undefined}
+ */
+function bucketDeleteReplication(authInfo, request, log, callback) {
+    log.debug('processing request', { method: 'bucketDeleteReplication' });
+    const { bucketName, headers, method } = request;
+    const metadataValParams = {
+        authInfo,
+        bucketName,
+        requestType: 'bucketOwnerAction',
+    };
+    return metadataValidateBucket(metadataValParams, log, (err, bucket) => {
+        const corsHeaders = collectCorsHeaders(headers.origin, method, bucket);
+        if (err) {
+            log.debug('error processing request', {
+                error: err,
+                method: 'bucketDeleteReplication',
+            });
+            return callback(err, corsHeaders);
+        }
+        if (!bucket.getReplicationConfiguration()) {
+            log.trace('no existing replication configuration', {
+                method: 'bucketDeleteReplication',
+            });
+            pushMetric('deleteBucketReplication', log, {
```

Comment:
philosophical question: should we count the no-op deletes in metrics?

- if the goal is to count the number of user operations executed, it's likely a yes
- if to count the actual actions that change a state, it would rather not be counted

Likely not counting those would give a counter semantic closer to the semantic of `putBucketReplication` counter (except you can re-write the policy multiple times but can only delete it once so they're not directly comparable).